.github/workflows/justtest.yaml

```diff
@@ -0,0 +1,699 @@
+#
+# Perform several builds in one workflow
+# --------------------------------------
+#
+# It seems that it doesn't complicate things too much.
+#
+# Split testing to jobs per test suite
+# ------------------------------------
+#
+# At first, it allows to restart a smaller piece of the workload
+# if an unstable test failed.
+#
+# Next, it allows to achieve larger parallelization level and
+# the feedback from CI is faster this way.
+#
+# Cache submodules build
+# ----------------------
+#
+# It is too attractive to have such a fast build. We use the
+# submodule hash to don't rebuild it each time.
+#
+# }}} Design choices
+#
+# TODO
+# ====
+#
+# This workflow is not ready to replace existing workflows that
+# perform the same build types. The reason is that
+# tarantool/luajit uses the existing workflows as reusable ones.
+
+name: justtest
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    # {{{ Build configurations
+
+    strategy:
+      matrix:
+        build-conf:
+          # TODO: Do we really need to run all the tests on the
+          # debug and release builds?
+          #
+          # First, there are tests that uses error injections,
+          # which are only available on the debug build.
+          #
+          # Second, the debug build has assertions enabled, so it
+          # can spot a problem that passes unseen on the release
+          # build.
+          #
+          # Third, the debug build has Lua's strict mode enabled
+          # by default.
+          #
+          # OTOH, the release build is what a customer typically
+          # uses, so it is the primary target. It feels
+          # counter-intuitive to verify the primary target with
+          # less tests that the additional targets.
+          #
+          # However, a release deployment job anyway performs the
+          # release build and it is different from this one in
+          # fact (at least the dependencies are linked statically
+          # and a specific building container is used). So, maybe,
+          # it is better to concentrate on developer needs in this
+          # workflow and reduce testing on the release build?
+          - debug
+          - release
+          # The idea is to verify that tarantool compiles with
+          # clang. At first, it is a code health check. And next,
+          # some developers use it.
+          #
+          # clang is known for its good analysis and warnings
+          # regarding potentially problematic code, but some of
+          # these nice linter capacities need analysis passes that
+          # are run on -O2, but not -O0. This way we can only get
+          # these warnings on RelWithDebInfo, but not on Debug.
+          #
+          # Seems like there is no reason to additionally verify
+          # the Debug build on clang.
+          #
+          # TODO: Whether we have to run all the tests with the
+          # clang build?
+          - clang
```

Comment:
The similar question here. It seems meaningful to verify that tarantool builds with clang. It seems good to check clang's diagnostics. However, what is the point to run all the tests here?